code/game/objects/effects/decals/cleanable/food.dm

```diff
@@ -38,8 +39,19 @@
 
 /obj/effect/decal/cleanable/food/salt/Bumped(atom/movable/AM)
 	. = ..()
-	if(is_species(AM, /datum/species/snail))
-		to_chat(AM, "<span class='danger'>Your path is obstructed by <span class='phobia'>salt</span>.</span>")
+
+	if(isliving(AM))
+
+		if(is_species(AM, /datum/species/snail))
+			to_chat(AM, "<span class='danger'>Your path is obstructed by <span class='phobia'>salt</span>.</span>")
```

Comment:
shouldn't this return here? otherwise, snails could destroy salt piles by just repeatedly running into them

Reply:
I fixed this by moving my code to Crossed()